test-case/manual/wov-script/wov-test.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+rm -rf blob
+preamble_time=0
+history_depth=0
+format=s16_le
+buffer_size=68000
+max_loop=100
+preamble_time_init=2100
+history_depth_init=1800
+str=$(amixer contents | grep "Detector")
+control=${str#*=}
+control_id=${control%%,*}
+
+
+usage(){
+        echo "Wrong parameters, please specify the test item"
+        echo "Eg: '-p' preamble_time, '-h' history_depth, '-f' format and '-s' period-size\
+        ./$0 -p 2500 -h 3000 -f s16_le -s 500"
+}
+
+# DEC to HEX
+dec2hex(){
+        printf "%x" "$1"
+}
+
+_save_def(){
+	echo "Save WoV as default configure blob:"
+	if [ ! -f detector-default-config ]; then
+		sudo sof-ctl -Dhw:0 -n "$control_id" -br -o detector-default-config
+	fi
+}
+
+_set_default(){
+	echo "Set WoV as default configure blob:"
+	sudo sof-ctl -Dhw:0 -n "$control_id" -br -s detector-default-config
+}
+
+# get the preamble time and history depth
+_get_pt_hd(){
+	_set_default
+        # save current WoV configure blob
+	sof-ctl -Dhw:0 -n "$control_id" -br -o blob
```

Comment:
Use longer, more descriptive names and an indirection variable so they can be changed easily if needed in the future.
```shell
 # just a suggestion, do not use these names, please find better names
wov_config=wov.txt

sof-ctl -Dhw:0 -n "$control_id" -br -o blob "$wov_config"
```